### PR TITLE
XW-6 | common check for crucial attributes of optimizely object

### PR DESCRIPTION
**Pull request by Warkot**

### CODE REVIEW

File: front/scripts/mercury/utils/variantTesting.ts [hunk 1]
```diff
@@ -43,6 +43,23 @@ module Mercury.Utils.VariantTesting {
 	}
 
 	/**
+	 * Checks if Optimizely object and its crucial data attributes are available
+	 *
+	 * @returns {boolean}
+	 */
+	export function isOptmizelyLoaded() {
```
Comment (rogatty) [inline front/scripts/mercury/utils/variantTesting.ts]: Typo, missing `i`

File: front/scripts/mercury/utils/variantTesting.ts [hunk 2]
```diff
@@ -43,6 +43,23 @@ module Mercury.Utils.VariantTesting {
...
+		var optimizely = window.optimizely;
+
+		return optimizely &&
+			optimizely.activeExperiments &&
+			optimizely.activeExperiments.length > 0 &&
+			typeof optimizely.allExperiments === 'object' &&
+			Object.keys(optimizely.allExperiments).length > 0 &&
+			typeof optimizely.variationNamesMap === 'object' &&
+			Object.keys(optimizely.variationNamesMap).length > 0;
```
Comment (rogatty) [inline front/scripts/mercury/utils/variantTesting.ts]: If we're checking for `length > 0` then maybe this function shouldn't be called `isOptimizelyLoaded` but `isOptmizelyLoadedAndActive` or something?

**Comment by rogatty** [comment]
LGTM

**Comment by vforge** [comment]
LGTM

File: front/scripts/mercury/utils/variantTesting.ts [hunk 3]
```diff
@@ -55,6 +73,8 @@ module Mercury.Utils.VariantTesting {
 			experimentName: string,
 			variationName: string;
 
+		// UA integration code is also used in MediaWiki app - if you change it here, change it there too:
+		// https://github.com/Wikia/app/blob/XW-6/extensions/wikia/AnalyticsEngine/js/universal_analytics.js#L333
```
Comment (rogatty) [inline front/scripts/mercury/utils/variantTesting.ts]: This link won't work after deleting the branch. Would be safer too give a link to file in dev branch and a name of the function.

**Pull request merged**